tests/unit/test_ecies.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2020 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "mococrw/ecies.h"
+#include "mococrw/key.h"
+#include "mococrw/openssl_wrap.h"
+#include "mococrw/util.h"
+#include "mococrw/symmetric_crypto.h"
+
+using namespace mococrw;
+using namespace ::testing;
+
+using testing::Eq;
+
+class ECIESTests : public ::testing::Test
+{
+public:
+    void SetUp() override
+    {
+        ECIESEncryptionCtxBuilder encBuilder(secp384PublicKey);
+        ECIESDecryptionCtxBuilder decBuilder(secp384Key);
+
+        const SymmetricCipherMode mode = SymmetricCipherMode::CBC;
+        const SymmetricCipherKeySize keySize = SymmetricCipherKeySize::S_256;
+        const SymmetricCipherPadding padding = SymmetricCipherPadding::PKCS;
+        const std::vector<uint8_t> zeroIV = std::vector<uint8_t>(AESCipherBuilder::getDefaultIVLength(mode));
+
+        kdf = std::shared_ptr<X963KDF>(new X963KDF(openssl::DigestTypes::SHA512));
```

Comment:
use of std::make_shared is preferred. See Effective Modern C++ Item 21 on why

Reply:
Done